assets/src/edit-story/components/panels/design/preset/useAddPreset.js

```diff
@@ -28,75 +28,101 @@ import {
   getTextPresets,
 } from './utils';
 
-function useAddPreset(presetType) {
-  const { currentPage, selectedElements, stylePresets, updateStory } = useStory(
+function useAddPreset(presetType, isLocal = false) {
+  const {
+    currentPage,
+    localColorPresets,
+    selectedElements,
+    stylePresets,
+    updateStory,
+  } = useStory(
     ({
       state: {
         currentPage,
         selectedElements,
-        story: { stylePresets },
+        story: { stylePresets, localColorPresets },
       },
       actions: { updateStory },
     }) => {
       return {
         currentPage,
+        localColorPresets,
         selectedElements,
         stylePresets,
         updateStory,
       };
     }
   );
   const { colors, textStyles } = stylePresets;
+  const { colors: localColors } = localColorPresets;
 
   const isText = areAllType('text', selectedElements);
   const isBackground = selectedElements[0].id === currentPage.elements[0].id;
   const handleAddPreset = useCallback(
     (evt) => {
       evt.stopPropagation();
-      let addedPresets = {
-        textStyles: [],
-        colors: [],
-      };
+      let addedPresets = isLocal
+        ? {
+            colors: [],
+          }
+        : {
+            textStyles: [],
+            colors: [],
+          };
+      const currentPresets = isLocal ? localColorPresets : stylePresets;
       if (isText) {
         addedPresets = {
           ...addedPresets,
-          ...getTextPresets(selectedElements, stylePresets, presetType),
+          ...getTextPresets(selectedElements, currentPresets, presetType),
         };
       } else if (isBackground) {
         addedPresets = {
           ...addedPresets,
-          ...getPagePreset(currentPage, stylePresets),
+          ...getPagePreset(currentPage, currentPresets),
         };
       } else {
         addedPresets = {
           ...addedPresets,
-          ...getShapePresets(selectedElements, stylePresets),
+          ...getShapePresets(selectedElements, currentPresets),
         };
       }
       if (
         addedPresets.colors?.length > 0 ||
         addedPresets.textStyles?.length > 0
       ) {
-        updateStory({
-          properties: {
-            stylePresets: {
-              textStyles: [...textStyles, ...addedPresets.textStyles],
-              colors: [...colors, ...addedPresets.colors],
+        if (isLocal) {
+          updateStory({
+            properties: {
+              localColorPresets: {
+                colors: [...localColors, ...addedPresets.colors],
+              },
+            },
+          });
+        } else {
+          updateStory({
+            properties: {
+              stylePresets: {
+                textStyles: [...textStyles, ...addedPresets.textStyles],
+                colors: [...colors, ...addedPresets.colors],
+              },
             },
-          },
-        });
+          });
```

Comment:
I'm not a big fan of this logic being located here. It's not easy to implement a reducer for the story properties currently, but it would be nice to have a cleaner API here like `removeLocalColor`, `removeGlobalColor`, `addLocalColor`, `addGlobalColor`, etc rather than send everything through `updateStory` and handle all the complexity locally rather than centrally.

Reply:
Refactored a bit, PTAL.